Refactor `AccountConnection` component.

tomalec opened the pull request:
### Current look

Account Connection card has four distinct states

- connected
- connected but no `accountData`/`id` (yet or at all)
- connection state unknown
- not connected

(see screenshots below)


### Changes proposed in this Pull Request:
I believe there are a few things we can improve from a UX perspective (and one DevX).
I propose to introduce the following changes:



- Show the Pinterest logo, regardless of the state, to reduce the number of moving parts.
- Show the "Disconnect" button even if there is no account data, or before it's loaded. To help users disconnect even if some account data gets corrupted, or simply without waiting for everything to load.
- Show placeholder instead of a spinner when we wait only for the account details.
- Set the minimum height of the card's content to `40px` = the height of the spinner, to reduce the difference between states, and reduce the layout shift when the spinner disappears and the content is loaded.
- Move the spinner to the middle, to let the user anticipate both things to appear: left-aligned text and right-aligned button.
- Make the code DRY, by moving repeated markup outside of conditions.





@shivanarrthine  I'd appreciate your opinion here :)


### Screenshots:

#### Before 

![before Screenshot from 2021-12-28 12-02-48](https://user-images.githubusercontent.com/17435/147581880-b41b9b20-ab57-46c3-9491-e45f4706c4c5.png)

#### After

![Screenshot from 2021-12-28 16-08-03](https://user-images.githubusercontent.com/17435/147581990-4d01f320-64f1-438c-856d-9c672d221dfe.png)


### Detailed test instructions:

1. Go to the `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fconnection`
2. Watch the loading state
3. Disconnect account
4. Watch connect state for a second before get redirected to welcome page (BTW, is that redirect desired here?)
5. Click "Get started" to go to `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding`
6. Connect it there,
7. Get back to the first step of onboarding stepper
7. See the connected state

- I was not able to reproduce the preloading of connection status other than programmatically, by setting `isConnected` to false manually in https://github.com/woocommerce/pinterest-for-woocommerce/blob/update/account-connection/assets/source/setup-guide/app/steps/SetupAccount.js#L174:L174


<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Tweak - Improved preloading state of Account Connection Card

tomalec committed:
Refactor `AccountConnection` component.

- Show the Pinterest logo, regardless of the state, to reduce the number of moving parts.
- Show the "Disconnect" button even if there is no account data, or before it's loaded. To help users disconnect even if some account data gets corrupted, or simply without waiting for everything to load.
- Show placeholder instead of a spinner when we wait only for the account details.
- Set the minimum height of the card's content to `40px` = the height of the spinner, to reduce the difference between states, and reduce the layout shift when the spinner disappears and the content is loaded.
- -Make the code DRY, by moving repeated markup outside of conditions.

tomalec committed:
Fix CSS linting - whitespace issues

eason9487 commented on assets/source/setup-guide/app/style.scss:
```diff
@@ -240,9 +257,24 @@ $root: ".woocommerce-setup-guide";
 
 				.connection-info {
+					// Set minimum height to the height of the spinner preloader.
+					min-height: calc(40px);
```
❔ 
Not sure why it needs `calc`? It's a leftover after finding the height of spinner is not exposed? 🤔

tomalec replied:
Thanks for catching this, yes that was a leftover. Fixed in https://github.com/woocommerce/pinterest-for-woocommerce/pull/312/commits/839abef87e4803abdc89becd3dc0cda239523974

eason9487 reviewed (approved):
LGTM. Left a comment for an unsure thing but it should be a non-blocking item no matter keep or remove it.

tomalec committed:
Remove useless `calc` from CSS.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/312#discussion_r776151383

shivanarrthine commented:
Hey @tomalec, this looks great! Definitely in agreement with the proposed changes - this would help improve the UX of the component and create some consistency between the states.

tomalec commented:
Thanks for the reviews :)
